scripts/generate_native_modules_specs.rb

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/ruby
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'tmpdir'
+
+# Handle Core Modules
+def codegen_core_modules(codegenPath)
+  reactNativePath = File.join(__dir__, "..")
+  srcsDir = File.join(reactNativePath, "Libraries")
+  libraryName = "FBReactNativeSpec"
+  outputDir = File.join(srcsDir, libraryName, libraryName)
+
+  codegen(codegenPath, srcsDir, libraryName, outputDir)
+end
+
+def codegen(codegenPath, srcsDir, libraryName, outputDir)
+  Dir.mktmpdir do |dir|
+    schemaFilePath = File.join(dir, "schema-#{libraryName}.json")
+    combineCliPath = File.join(codegenPath, "lib", "cli", "combine", "combine-js-to-schema-cli.js")
+    generateSpecsCliPath = File.join(__dir__, "generate-native-modules-specs-cli.js")
+    system("node #{combineCliPath} #{schemaFilePath} #{srcsDir}") or raise "Could not generate Native Module schema"
+    system("node #{generateSpecsCliPath} ios #{schemaFilePath} #{outputDir} #{libraryName}") or raise "Could not generate code for #{libraryName}"
+  end
+end
+
+def main()
+  codegenPath = ENV["CODEGEN_PATH"] ||= File.join(reactNativePath, "..", "react-native-codegen")
+  codegen_core_modules(codegenPath)
+end
+
+main()
```

Comment:
Right now, this script will only handle the core native modules in Libraries/. It could easily be extended in a future PR to handle native modules in arbitrary locations (e.g. RNTester's own native modules, third party modules...)

Reply:
Also noting that [in the future] there is no reason that this script should remain in Ruby instead of JS or shell script.

Reply:
Good point. I'd rather not add a new script, so I'll use the existing bash script with some minor changes to allow consuming the codegen from an arbitrary path.